modules/ducktests/tests/docker/Dockerfile

```diff
@@ -48,14 +48,13 @@ ARG APACHE_MIRROR="https://apache-mirror.rbc.ru/pub/apache/"
 ARG APACHE_ARCHIVE="https://archive.apache.org/dist/"
 
 # Install binary test dependencies.
-ARG IGNITE_VERSION="2.8.0"
-ARG IGNITE_NAME="ignite-$IGNITE_VERSION"
-ARG IGNITE_RELEASE_NAME="apache-ignite-$IGNITE_VERSION-bin"
-
-ADD $APACHE_ARCHIVE/ignite/$IGNITE_VERSION/$IGNITE_RELEASE_NAME.zip /opt/
-RUN cd /opt && unzip $IGNITE_RELEASE_NAME.zip && rm $IGNITE_RELEASE_NAME.zip
-RUN mv /opt/$IGNITE_RELEASE_NAME /opt/$IGNITE_NAME
-RUN chmod a+wr /opt/$IGNITE_NAME -R
+ADD $APACHE_ARCHIVE/ignite/2.7.6/apache-ignite-2.7.6-bin.zip /opt/
+RUN cd /opt && unzip apache-ignite-2.7.6-bin.zip && mv /opt/apache-ignite-2.7.6-bin /opt/ignite-2.7.6 && chmod a+wr /opt/$IGNITE_NAME -R
+
+ADD $APACHE_ARCHIVE/ignite/2.8.0/apache-ignite-2.8.0-bin.zip /opt/
```

Comment:
Versions and distributive sources should be configured externally: easy add new versions, custom versions from custom sources, without changing dockerfile. 

Possible just try use different images for different versions (to save time on prepare tests to start)

Reply:
> Versions and distributive sources should be configured externally

Please, clarify, what issues do you see with the approach when we have all the required versions in this `Dockerfile`?
If we have to use a different image with some custom Ignite assembly, we, probably, should have script parameters to specify other `Dockerfile` to build image from.

Reply:
I see at least 3 issues there:

1. Some versions could be incompatible with each other. For example, one version could work on java11 without issues, while others are not (there is also issue with our custom build that works only on patched java). Accidentally there could be dependencies on different system libs, envs, directories, etc.

I think its a good practice do not mix different versions in the same env and prepare a separate image for every version. 

2. Also there should be images like ignite-X.Y.Z-java8, ignite-X.Y.Z-java11, ignite-X.Y.Z-alpine, etc. I think we can reuse this common images in tests rather than creating uber image just for tests.

3. I'm trying to write some ISE specific tests with enabled custom plugins. Also I want to run existing tests (written for original Ignite) for the custom version directly from the ducktape directory. Also I want compare multiple versions (custom and original) with each other.

I suppose that I can flexible configure my test just specify required versions rather than re-create new image for my every case.

Reply:
> For example, one version could work on java11 without issues, while others are not

This knowledge should be put in the test itself. Kafka uses the same approach.

> Accidentally there could be dependencies on different system libs, envs, directories, etc.

All required dependencies exist in zip distribution.
Directories structure doesn't change frequently so we can put this knowledge into the service itself.

> I think its a good practice do not mix different versions

What is wrong with the different versions?
We want to run the same tests on the different versions.

> Also there should be images like ignite-X.Y.Z-java8, ignite-X.Y.Z-java11, ignite-X.Y.Z-alpine

1. We don't support alpine.
2. Yes, we can parametrize the creation of the image with the JDK version. Initial PR doesn't contain this parameter for the sake of simplicity.

> I'm trying to write some ISE specific tests with enabled custom plugins
> rather than re-create new image for my every case.

You should create different docker image to be able to use custom Ignite assembly.
You don't need several images, just one image that satisfies your requirements.

Reply:
> All required dependencies exist in zip distribution.
Actually this is not true as at least Spark set up separately. And there could be issues with different versions of Ignite and Spark.

With storing dependencies in .zip you just reinvent containers without using any of their advantages. Now you need to support multiple applications in one container. But you can't start all of them in one container as you need to orchestrate apps by self: set LD_LIBRARY_PATH for different system libs, set JAVA_HOME for different jvms, use different networks, different mount directories, etc.

If you move JDK as param for container why do not do the same for Ignite version? It looks inconsistent now.
> Kafka uses the same approach
Kafka may have a motivation for that. To do the same one need clearly understand reasons behind this approach. Could you please provide discussion or design doc that describes this solution.

> We don't support alpine.
It's ok, it was just an example that there are should be different public official images that can be reused. I'm repeating there we could use public Ignite images to use them in tests instead of creating uber docker image just for tests. 

> You don't need several images, just one image that satisfies your requirements
Containers are building blocks. A container provides _a process_ and it is not a replacement of physical environment. There are additional limitations if try use containers as envs. For example in current dockerfile at least 2 problems that could lead to wasting time:
1. _delegated_ volumes I've mentioned below (check this [comment](https://github.com/apache/ignite/pull/7967#discussion_r449182086));
2. Using 'tail -f' as init process could lead to wasting Linux resources (Check this [article](https://blog.phusion.nl/2015/01/20/docker-and-the-pid-1-zombie-reaping-problem/)).

Reply:
> Actually this is not true as at least Spark set up separately. And there could be issues with different versions of Ignite and Spark.

We just should prepare all Spark or Ignite versions we want to tests.
I can't see any issues with this.

Reply:
> If you move JDK as param for container why do not do the same for Ignite version? It looks inconsistent now.

Because I think we don't want to increase the complexity of the final solution with dozens of different images.

Reply:
> It's ok, it was just an example that there are should be different public official images that can be reused

For now, we don't have those images.
Moreover, we want to use the same tests with the different deployment types - AWS, private cloud, etc.
If we go with the dozens of images we should provide a dozen scripts for each deployment type, also.

Reply:
Lets, restart this thread at separate PR if necessary?